railties/test/application/asset_debugging_test.rb

```diff
@@ -68,5 +68,102 @@ class ::PostsController < ActionController::Base ; end
       assert_match(/<script src="\/assets\/application(\.self)?-([0-z]+)\.js\?body=1"><\/script>/, last_response.body)
       assert_match(/<script src="\/assets\/xmlhr(\.self)?-([0-z]+)\.js\?body=1"><\/script>/, last_response.body)
     end
+
+    test "public path and tag methods are not over-written by the asset pipeline" do
+      contents = "doesnotexist"
+      cases = {
+        asset_path:             %r{/#{ contents }},
+        image_path:             %r{/images/#{ contents }},
+        video_path:             %r{/videos/#{ contents }},
+        audio_path:             %r{/audios/#{ contents }},
+        font_path:              %r{/fonts/#{ contents }},
+        javascript_path:        %r{/javascripts/#{ contents }},
+        stylesheet_path:        %r{/stylesheets/#{ contents }},
+        image_tag:              %r{<img src="/images/#{ contents }"},
+        favicon_link_tag:       %r{<link rel="shortcut icon" type="image/x-icon" href="/images/#{ contents }" />},
+        stylesheet_link_tag:    %r{<link rel="stylesheet" media="screen" href="/stylesheets/#{ contents }.css" />},
+        javascript_include_tag: %r{<script src="/javascripts/#{ contents }.js">},
+        audio_tag:              %r{<audio src="/audios/#{ contents }"></audio>},
+        video_tag:              %r{<video src="/videos/#{ contents }"></video>}
+      }
+
+      cases.each do |(view_method, tag_match)|
+        app_file "app/views/posts/index.html.erb", "<%= #{ view_method } '#{contents}', skip_pipeline: true %>"
```

Comment:
Mixed spaces around interpolation braces abound in this test file. I think our preferred style is to not have any around interpolation braces, e.g.: `#{view_method}`.